core/domain/question_services.py

```diff
@@ -217,3 +224,254 @@ def update_question(
+        creator_id, question.id, question_content)
+
+    return question_summary
+
+
+def save_question_summary(question_summary):
+    """Save a question summary domain object as a QuestionSummaryModel
+    entity in the datastore.
+
+    Args:
+        question_summary: The question summary object to be saved in the
+            datastore.
+    """
+    question_summary_model = question_models.QuestionSummaryModel(
+        id=question_summary.id,
+        creator_id=question_summary.creator_id,
+        question_model_last_updated=question_summary.last_updated,
+        question_model_created_on=question_summary.created_on,
+        question_content=question_summary.question_content
+    )
+
+    question_summary_model.put()
+
+
+def get_question_summary_from_model(question_summary_model):
+    """Returns a domain object for an Oppia question summary given a
+    questioin summary model.
+
+    Args:
+        question_summary_model: QuestionSummaryModel.
+
+    Returns:
+        QuestionSummary.
+    """
+    return question_domain.QuestionSummary(
+        question_summary_model.creator_id,
+        question_summary_model.id,
+        question_summary_model.question_content,
+        question_summary_model.question_model_created_on,
+        question_summary_model.question_model_last_updated
+    )
+
+
+def get_question_summaries_by_creator_id(creator_id):
+    """Gets question summaries of questions created by the user.
+
+    Args:
+        creator_id: str. The user ID of the creator.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    question_summary_models = (
+        question_models.QuestionSummaryModel.get_by_creator_id(creator_id))
+
+    for question_summary_model in question_summary_models:
+        question_summaries = [
+            get_question_summary_from_model(question_summary_model)]
+
+    return question_summaries
+
+
+def get_question_summary_by_question_id(question_id, strict=False):
+    """Gets question summary of question by ID of the question.
+
+    Args:
+        question_id: str. The ID of the question.
+        strict: bool. Whether to fail noisily if no question summary for the
+            given question id exists in the datastore.
+
+    Returns:
+        QuestionSummaryModel. The QuestionSummaryModel for the given question.
+    """
+    return get_question_summary_from_model(
+        question_models.QuestionSummaryModel.get(question_id, strict=strict))
+
+
+def get_skill_ids_linked_with_question(question_id):
+    """Gets linked skill IDs for given question.
+
+    Args:
+        question_id: str. The question ID for the given question.
+
+    Returns:
+        list(SkillSummary)|None. The list of skill summaries for all
+            linked skill ids or None if there exists no skill linked
+            for question.
+    """
+    question_skill_links = question_models.QuestionSkillLinkModel.get(
```

Comment:
Why "links" (plural) if you're getting just one model? Call it question_skill_link_model

Reply:
In fact something seems completely wrong with this function. Somehow a get() has turned into a multi-model query...

Also, I realized this function has no tests. Make sure all functions in this file are tested.